components/brave_rewards/browser/rewards_service_impl.h

```diff
@@ -346,6 +347,18 @@ class RewardsServiceImpl : public RewardsService,
 
   bool IsRewardsEnabled() const override;
 
+  bool GetScheduledCaptchaInfo(std::string* captcha_url,
```

Comment:
Adding this to rewards service seems like a bad idea to me. This class is already ridiculously bloated, wouldn't it be better as a separate keyed service?

Reply:
I mean, this seems like it would be much simpler if you made a separate service now, but I guess I'll leave it up to you guys to do now or follow-up.

Reply:
in particular the buildflags would go away (which probably aren't needed anyway if you remove the content dependency as discussed)

Reply:
actually after looking at the AdsServiceImpl code I take that back. This is creating a mess of issues with duplicated methods in AdsServiceImpl. See https://github.com/brave/brave-core/pull/9350/files#r694381232

Reply:
Reworked to make `BraveAdaptiveCaptchaService` a keyed service.